<a href="https://colab.research.google.com/github/Gtebaldi1/3-T-tests/blob/main/HHA_507_T_test_assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [20]:
import pandas as pd
from scipy.stats import ttest_ind
from numpy.random import seed
from numpy.random import randn
import scipy.stats as stats
from scipy.stats import shapiro
from matplotlib import pyplot

## Add data frame

In [39]:
diabetes = pd.read_csv('https://raw.githubusercontent.com/hantswilliams/AHI_DataSci_507/main/Datasets/Diabetes/DB1_Diabetes/diabetic_data.csv')

## variables in data frame

In [29]:
list(diabetes)

['encounter_id',
 'patient_nbr',
 'race',
 'gender',
 'age',
 'weight',
 'admission_type_id',
 'discharge_disposition_id',
 'admission_source_id',
 'time_in_hospital',
 'payer_code',
 'medical_specialty',
 'num_lab_procedures',
 'num_procedures',
 'num_medications',
 'number_outpatient',
 'number_emergency',
 'number_inpatient',
 'diag_1',
 'diag_2',
 'diag_3',
 'number_diagnoses',
 'max_glu_serum',
 'A1Cresult',
 'metformin',
 'repaglinide',
 'nateglinide',
 'chlorpropamide',
 'glimepiride',
 'acetohexamide',
 'glipizide',
 'glyburide',
 'tolbutamide',
 'pioglitazone',
 'rosiglitazone',
 'acarbose',
 'miglitol',
 'troglitazone',
 'tolazamide',
 'examide',
 'citoglipton',
 'insulin',
 'glyburide-metformin',
 'glipizide-metformin',
 'glimepiride-pioglitazone',
 'metformin-rosiglitazone',
 'metformin-pioglitazone',
 'change',
 'diabetesMed',
 'readmitted']

## T-TESTS
## Question 1: is there a difference between sex (M:F) and the number of days in hospital?

In [40]:
Female = diabetes[diabetes['gender'] == 'Female']
Male = diabetes[diabetes['gender'] == 'Male']

In [41]:
ttest_ind(Female['time_in_hospital'], Male['time_in_hospital'])

Ttest_indResult(statistic=9.542637042242887, pvalue=1.4217299655114968e-21)

##Results:
##statistic=9.542637042242887
##pvalue=1.4217299655114968e-21)
## the p value is <.05 this indicates there is a significant difference between the two groups, between females and males and number of days each gender spent in hospital

##Question 2: is there a difference between RACE (Caucasian and African American) and the number of days in hospital?

In [31]:
Caucasian = diabetes[diabetes['race'] == 'Caucasian'] 
African_American = diabetes[diabetes['race'] == 'AfricanAmerican']

In [33]:
ttest_ind(Caucasian['time_in_hospital'], African_American ['time_in_hospital'])

Ttest_indResult(statistic=-5.0610017032095325, pvalue=4.178330085585203e-07)

##Resulst:
##statistic=-5.0610017032095325 
##pvalue=4.178330085585203e-07
## the p value is <.05 which indicates there is a significant difference between the two groups, race ( caucasian and african american) and the number of days in the hospital

##Question 3:is there a difference between RACE (Asian and African American) and the number of lab procedures performed?

In [45]:
Asian = diabetes[diabetes['race'] == 'Asian']
African_American = diabetes[diabetes['race'] == 'AfricanAmerican']

In [46]:
ttest_ind(Asian['num_lab_procedures'], African_American['num_lab_procedures'])

Ttest_indResult(statistic=-3.9788715315360292, pvalue=6.948907528800307e-05)

##Results: 
##statistic=-3.9788715315360292
##pvalue=6.948907528800307e-05
## the p value is <.05 which indicates there is a significant difference between the two groups, race ( asian and african american) and number of days spent in the hospital
